### PR TITLE
[NEW][APPS] Possibility to set room closer via Apps LivechatBridge.closeRoom

### PR DESCRIPTION
**Pull request by cuonghuunguyen**
## Proposed changes (including videos or screenshots)

Add an optional param named `closer` into `LivechatBridge.closeRoom` so that it will be possible to close the room and send a close room message with the correct room closer.
If the param is not passed, use the room visitor as the room closer.

## Issue(s)
closes RocketChat/Rocket.Chat.Apps-Engine#373

Apps-Engine's PR: RocketChat/Rocket.Chat.Apps-Engine#391

### CODE REVIEW

File: app/apps/server/bridges/livechat.js [hunk 1]
```diff
@@ -72,14 +72,21 @@ export class AppLivechatBridge {
...
 
-		return Livechat.closeRoom({
-			visitor: this.orch.getConverters().get('visitors').convertAppVisitor(room.visitor),
+		const closeData = {
 			room: this.orch.getConverters().get('rooms').convertAppRoom(room),
 			comment,
-		});
+		};
+
+		if (closer?.id) {
```
Comment (lucassartor) [inline app/apps/server/bridges/livechat.js:83]: Hey @renatobecker! Can you give me some help? 
I'm not really familiar with the Livechat codebase. Do you think this is the right approach in this situation, sending the `visitor` and `user` in different situations?

Comment (lucassartor) [inline app/apps/server/bridges/livechat.js:83]: Hey @renatobecker! Can you give me some help? 
I'm not really familiar with the Livechat codebase. Do you think this is the right approach in this situation, sending the `visitor` and `user` in different situations?

File: app/apps/server/bridges/livechat.ts [hunk 2]
```diff
@@ -87,15 +87,21 @@ export class AppLivechatBridge extends LivechatBridge {
...
-		return Livechat.closeRoom({
-			visitor: this.orch.getConverters()?.get('visitors').convertAppVisitor(room.visitor),
+		const closeData: any = {
 			room: this.orch.getConverters()?.get('rooms').convertAppRoom(room),
 			comment,
-			user: undefined,
-		});
+		};
+
+		if (closer?.id) {
```
Comment (renatobecker) [inline app/apps/server/bridges/livechat.ts:98]: You don't need this `if` condition here, you can rely on conditional properties with object spread like the code below:

```
const shouldAddProp2 = false;
const obj = {
  prop1: 2,
  ...shouldAddProp2 && { conditionalProp2: 2 }
}
```
This means that you can implement the logic inside the [closeData](https://github.com/RocketChat/Rocket.Chat/pull/21025/files#diff-0d545a42ed0213be9e61429689f3d0c2012da94d7a7ba988dffd3a02c10b3557R93-R96) object.

File: app/apps/server/bridges/livechat.ts [hunk 3]
```diff
@@ -87,15 +87,21 @@ export class AppLivechatBridge extends LivechatBridge {
...
+			closeData.user = this.orch.getConverters()?.get('users').convertById(closer.id);
```
Comment (renatobecker) [inline app/apps/server/bridges/livechat.ts:99]: Where are you checking if the user with `close.id` exists? What if it doesn't?

**Review by renatobecker — CHANGES_REQUESTED** [review]

File: app/apps/server/bridges/livechat.ts (hunk 2 above)
Comment (cuonghuunguyen) [reply app/apps/server/bridges/livechat.ts]: applied suggestion

File: app/apps/server/bridges/livechat.ts (hunk 3 above)
Comment (cuonghuunguyen) [reply app/apps/server/bridges/livechat.ts]: solved by the fix above

**Review by d-gubert — APPROVED** [review]
Thank you for the contribution!!

**Review by d-gubert — APPROVED** [review]
For realz now, thank you for the contribution!! :D

**Pull request merged**